ring/ring.go

```diff
@@ -286,14 +286,13 @@ func (r *Ring) starting(ctx context.Context) error {
 func (r *Ring) loop(ctx context.Context) error {
 	// Update the ring metrics at start of the main loop.
 	r.updateRingMetrics()
-	go func() {
-		// Start metrics update ticker to update the ring metrics.
-		ticker := time.NewTicker(10 * time.Second)
-		defer ticker.Stop()
 
+	metricChan := make(chan struct{}, 1)
+	go func() {
 		for {
 			select {
-			case <-ticker.C:
+			case <-metricChan:
+				time.After(time.Second) // debounce and avoid consuming too much CPU
```

Comment:
This is not a sleep. You need to wait for a message on the returned channel (and also handle `ctx.Done`). See example in doc:
https://pkg.go.dev/time#After

Reply:
went with different solution